### PR TITLE
Starts trying to convert scrubbers to XGM

### PR DESCRIPTION
**Pull request by MadmanMartian**
Great googly moogly are mapped scrubbers trash.

Did you know that half of box security corridor has scrubbers set to not syphon plasma, simply because of mapping accidents?

### CODE REVIEW

File: code/game/machinery/alarm.dm [hunk 1]
```diff
@@ -109,24 +109,24 @@
 			AA.preset=preset
 			apply_preset(1) // Only this air alarm should send a cycle.
 
-	TLV["oxygen"] =			list(16, 19, 135, 140) // Partial pressure, kpa
-	TLV["nitrogen"] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
-	TLV["carbon_dioxide"] = list(-1.0, -1.0, 5, 10) // Partial pressure, kpa
-	TLV["plasma"] =			list(-1.0, -1.0, 0.2, 0.5) // Partial pressure, kpa
-	TLV["n2o"] =			list(-1.0, -1.0, 0.5, 1.0) // Partial pressure, kpa
+	TLV[GAS_OXYGEN] =			list(16, 19, 135, 140) // Partial pressure, kpa
+	TLV[GAS_OXYGEN] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	TLV[GAS_NITROGEN] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
```

File: code/game/machinery/alarm.dm [hunk 2]
```diff
@@ -109,24 +109,24 @@
...
+	TLV[GAS_SLEEPING] =			list(-1.0, -1.0, 0.5, 1.0) // Partial pressure, kpa
 	TLV["other"] =			list(-1.0, -1.0, 0.5, 1.0) // Partial pressure, kpa
 	TLV["pressure"] =		list(ONE_ATMOSPHERE*0.80,ONE_ATMOSPHERE*0.90,ONE_ATMOSPHERE*1.10,ONE_ATMOSPHERE*1.20) /* kpa */
 	TLV["temperature"] =	list(T0C-30, T0C, T0C+40, T0C+70) // K
 	target_temperature = T0C+20
 	switch(preset)
 		if(AALARM_PRESET_VOX) // Same as usual, s/nitrogen/oxygen
-			TLV["nitrogen"] = 		list(16, 19, 135, 140) // Vox use same partial pressure values for N2 as humans do for O2.
-			TLV["oxygen"] =			list(-1.0, -1.0, 0.5, 1.0) // Under 1 kPa (PP), vox don't notice squat (vox_oxygen_max)
+			TLV[GAS_OXYGEN] = 		list(16, 19, 135, 140) // Vox use same partial pressure values for N2 as humans do for O2.
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
			TLV[GAS_NITROGEN] = 		list(16, 19, 135, 140) // Vox use same partial pressure values for N2 as humans do for O2.
```

File: code/game/machinery/alarm.dm [hunk 3]
```diff
@@ -176,10 +176,10 @@
 
 	// breathable air according to human/Life()
 	/*
-	TLV["oxygen"] =			list(16, 19, 135, 140) // Partial pressure, kpa
-	TLV["nitrogen"] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
-	TLV["carbon_dioxide"] = list(-1.0, -1.0, 5, 10) // Partial pressure, kpa
-	TLV["plasma"] =			list(-1.0, -1.0, 0.2, 0.5) // Partial pressure, kpa
+	TLV[GAS_OXYGEN] =			list(16, 19, 135, 140) // Partial pressure, kpa
+	TLV[GAS_OXYGEN] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	TLV[GAS_NITROGEN] =		list(-1, -1,  -1,  -1) // Partial pressure, kpa
```

File: code/game/machinery/alarm.dm [hunk 4]
```diff
@@ -291,12 +291,12 @@
 
 
 	var/pressure_dangerlevel = get_danger_level(environment.pressure, TLV["pressure"])
-	var/oxygen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_OXYGEN), TLV["oxygen"])
-	var/nitrogen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_NITROGEN), TLV["nitrogen"])
-	var/co2_dangerlevel = get_danger_level(environment.partial_pressure(GAS_CARBON), TLV["carbon_dioxide"])
-	var/plasma_dangerlevel = get_danger_level(environment.partial_pressure(GAS_PLASMA), TLV["plasma"])
+	var/oxygen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_OXYGEN), TLV[GAS_OXYGEN])
+	var/nitrogen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_NITROGEN), TLV[GAS_OXYGEN])
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	var/nitrogen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_NITROGEN), TLV[GAS_NITROGEN])
```

File: code/game/machinery/alarm.dm [hunk 5]
```diff
@@ -535,23 +535,23 @@
 	var/list/current_settings = TLV["pressure"]
 	var/pressure_dangerlevel = get_danger_level(environment.pressure, current_settings)
 
-	current_settings = TLV["oxygen"]
+	current_settings = TLV[GAS_OXYGEN]
 	var/oxygen_dangerlevel = get_danger_level(environment.partial_pressure(GAS_OXYGEN), current_settings)
 	var/oxygen_percent = round(environment[GAS_OXYGEN] / total * 100, 2)
 
-	current_settings = TLV["nitrogen"]
+	current_settings = TLV[GAS_OXYGEN]
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	current_settings = TLV[GAS_NITROGEN]
```

File: code/game/machinery/alarm.dm [hunk 6]
```diff
@@ -576,22 +576,22 @@
 	data["temperature_c"]=round(environment.temperature - T0C, 0.1)
 
 	var/percentages[0]
-	percentages["oxygen"]=oxygen_percent
-	percentages["nitrogen"]=nitrogen_percent
+	percentages[GAS_OXYGEN]=oxygen_percent
+	percentages[GAS_OXYGEN]=nitrogen_percent
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	percentages[GAS_NITROGEN]=nitrogen_percent
```

File: code/game/machinery/alarm.dm [hunk 7]
```diff
@@ -576,22 +576,22 @@
...
 	percentages["other"]=other_percent
 	data["contents"]=percentages
 
 	var/danger[0]
 	danger["pressure"]=pressure_dangerlevel
 	danger["temperature"]=temperature_dangerlevel
-	danger["oxygen"]=oxygen_dangerlevel
-	danger["nitrogen"]=nitrogen_dangerlevel
+	danger[GAS_OXYGEN]=oxygen_dangerlevel
+	danger[GAS_OXYGEN]=nitrogen_dangerlevel
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]:
```suggestion
	danger[GAS_NITROGEN]=nitrogen_dangerlevel
```

File: code/game/machinery/alarm.dm [hunk 8]
```diff
@@ -576,22 +576,22 @@
 	data["temperature_c"]=round(environment.temperature - T0C, 0.1)
 
 	var/percentages[0]
-	percentages["oxygen"]=oxygen_percent
-	percentages["nitrogen"]=nitrogen_percent
+	percentages[GAS_OXYGEN]=oxygen_percent
+	percentages[GAS_OXYGEN]=nitrogen_percent
 	percentages["co2"]=co2_percent
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]: I'm not sure if you'd also need to update something somewhere else if you changed this to `GAS_CARBON`, but if not you should

File: code/game/machinery/alarm.dm [hunk 9]
```diff
@@ -576,22 +576,22 @@
...
 	data["contents"]=percentages
 
 	var/danger[0]
 	danger["pressure"]=pressure_dangerlevel
 	danger["temperature"]=temperature_dangerlevel
-	danger["oxygen"]=oxygen_dangerlevel
-	danger["nitrogen"]=nitrogen_dangerlevel
+	danger[GAS_OXYGEN]=oxygen_dangerlevel
+	danger[GAS_OXYGEN]=nitrogen_dangerlevel
 	danger["co2"]=co2_dangerlevel
```
Comment (Exxion) [inline code/game/machinery/alarm.dm]: Same here as the other one that isn't a suggested change

**Review by Exxion — CHANGES_REQUESTED** [review]
I think you made a mistake on one thing when using find and replace

**Review by Exxion — APPROVED** [review]

**Comment by PJB3005** [comment]
This broke air alarm scrubber preferences it seems.

**Comment by MadmanMartian** [comment]
It did, because it was far from finished.

**Comment by PJB3005** [comment]
Mark the PR as explicitly WiP next time.

**Comment by SirSkeleto** [comment]
or you could use github's cool new draft pr feature

**Comment by PJB3005** [comment]
Yes, that's what I meant.

**Pull request merged**